crates/astria-sequencer/src/mempool.rs

```diff
@@ -117,6 +133,81 @@ impl std::hash::Hash for EnqueuedTransaction {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct TransactionExpired;
+
+impl std::fmt::Display for TransactionExpired {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Transaction expired")
+    }
+}
+
+const TX_TTL: Duration = Duration::from_secs(600); // 10 minutes 
+const REMOVAL_CACHE_SIZE: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(4096) }; // TODO make configuration variable 
+const TX_REMOVAL_TTL: Duration = Duration::from_secs(30); // 30 seconds
+
+/// `RemovalCache` is a cache used for signaling to `CometBFT` when a
+/// transaction can be removed from the `CometBFT` mempool outside of the
+/// `CheckTx` checks.
+///
+/// This is useful for when a transaction fails execution or when a transaction
+/// has expired in the app's mempool.
+///
+/// The cache places a TTL on the entries, allowing users
+/// to request their transaction to be processed again after the
+/// ttl has expired. This is useful for users who want to re-run a transaction
+/// that failed to execute or to re-add a transaction that has expired.
+#[derive(Clone)]
+pub(crate) struct RemovalCache {
+    cache: HashMap<[u8; 32], (Instant, Arc<anyhow::Error>)>,
```

Comment:
for the error, the only two options right now are expired and failed execution right? i'd prefer to change the `Arc<anyhow::Error>` to an enum, that would be clearer as to which removal reasons exist

Reply:
I was trying to bubble up the reason for failure to the user, but it gets stuck at 'failed execution' or 'failed stateful checks'. I can make it an enum if you don't think that the `CheckTx` failure case should be more specific than 'failed execution'